truckersmp-cli

```diff
@@ -292,55 +315,67 @@ Please report an issue: {}""".format(e, URL.issueurl))
                             break
                         md5hash.update(buf)
                 if md5hash.hexdigest() != md5:
-                    files_to_download.append(("/files" + jsonfilepath, modfilepath, md5))
+                    dlfiles.append(("/files" + jsonfilepath, modfilepath, md5))
             except Exception as e:
                 sys.exit("Failed to read {}: {}".format(modfilepath, e))
-    if len(files_to_download) > 0:
+    if len(dlfiles) > 0:
         printdbg("Files to download:")
-        for path, _, _ in files_to_download:
+        for path, _, _ in dlfiles:
             printdbg("  {}".format(path))
     else:
         printdbg("No files to download")
 
     # download missing/wrong files
-    if not download_files(URL.dlurl, files_to_download):
-        if not download_files(URL.dlurlalt, files_to_download):
+    if not download_files(URL.dlurl, dlfiles):
+        if not download_files(URL.dlurlalt, dlfiles):
             # something went wrong
             sys.exit("Failed to download mod files.")
 
+
 def update_game():
     # make sure steam is closed before updating
     # it's possible to update with the steam client open but the client looses
     # all connectivity and asks for password and steam guard code after restart
-    while True:
-        try:
-            subprocess.check_call(["pidof", "steam"], stdout = subprocess.DEVNULL)
-        except:
-            break
-        else:
-            printdbg("Found running steam, running pkill -x steam")
-            subprocess.call(["pkill", "-x", "steam"])
-            time.sleep(5)
+    try:
+        # space-separated list of steam pids
+        steam_pids_str = str(
+          subproc.check_output(["pidof", "steam"]), "ascii").rstrip()
+        # list of steam pids (strings)
+        steam_pid_list = steam_pids_str.split(" ")
+    except Exception:
+        pass
+    else:
+        printdbg("Found running steam, killing the processes")
+        # number of processes to kill
+        process_count = len(steam_pid_list)
+        # wait until all processes are terminated
+        while process_count > 0:
+            try:
+                for pid in steam_pid_list:
+                    os.kill(int(pid), signal.SIGTERM)
+            except OSError as e:
+                if e.errno == errno.ESRCH:
+                    process_count -= 1
```

Comment:
If another user has an instance of steam open, this will loop infinitely. We could either check for `errno.EPERM` or use `pgrep -u $USER -x steam` to limit the PID search to those owned by the user.